components/script/dom/browsingcontext.rs

```diff
@@ -104,10 +114,10 @@ impl BrowsingContext {
     /// Change the Window that this browsing context's WindowProxy resolves to.
     // TODO: support setting the window proxy to a dummy value,
     // to handle the case when the active document is in another script thread.
-    pub fn set_window_proxy(&self, window: &Window) {
+    fn set_window_proxy(&self, window: &GlobalScope, traps: &ProxyTraps) {
         unsafe {
             debug!("Setting window proxy of {:p}.", self);
-            let WindowProxyHandler(handler) = window.windowproxy_handler();
+            let handler = CreateWrapperProxyHandler(traps);
```

Comment:
We shouldn't need to create a new handler every time here. We can create it once and store it in DomStaticData like the [regular](https://dxr.mozilla.org/servo/rev/287b02e21fa2c81d58b070be36add5e951512679/components/script/dom/bindings/utils.rs#61) windowproxy handler.